### DISCUSSION TITLE
How to set one frame per second for sprites？

### DISCUSSION THREAD

**Discussion by nikitazep** [mediacms-io/mediacms#949] [Q&A]
I found this code about sprites in /files/tasks.py, can you explain how the code works?

@task(name="produce_sprite_from_video", queue="long_tasks")
def produce_sprite_from_video(friendly_token):
    """Produces a sprites file for a video, uses ffmpeg"""

    try:
        media = Media.objects.get(friendly_token=friendly_token)
    except BaseException:
        logger.info("failed to get media with friendly_token %s" % friendly_token)
        return False

    with tempfile.TemporaryDirectory(dir=settings.TEMP_DIRECTORY) as tmpdirname:
        try:
            tmpdir_image_files = tmpdirname + "/img%03d.jpg"
            output_name = tmpdirname + "/sprites.jpg"
            cmd = "{0} -i {1} -f image2 -vf 'fps=1/10, scale=160:90' {2}&&files=$(ls {3}/img*.jpg | sort -t '-' -n -k 2 | tr '\n' ' ')&&convert $files -append {4}".format(
                settings.FFMPEG_COMMAND,
                media.media_file.path,
                tmpdir_image_files,
                tmpdirname,
                output_name,
            )
            subprocess.run(cmd, stdout=subprocess.PIPE, shell=True)
            if os.path.exists(output_name) and get_file_type(output_name) == "image":
                with open(output_name, "rb") as f:
                    myfile = File(f)
                    media.sprites.save(
                        content=myfile,
                        name=get_file_name(media.media_file.path) + "sprites.jpg",
                    )
        except BaseException:
            pass
    return True

**Comment by mgogoulos** [comment]
sprites is a FileField on Media (the basic MediaCMS Django models that stores info about all files)
so the celery task here is performing the generation of sprites, and saves it as a FileField upon completion. This task can take time, for big videos, this is why it is run on the long_tasks queue. 

So what it does is generate a command, and then executes this command. The command is ffmpeg mostly, which generates the small images, and then using `convert` and some bash scripting it is making a .jpg file. I don't remember the ffmpeg documentation for making that change, but I'm pretty sure you can make adjustments easily if you change code on that place!

**Comment by mgogoulos** [comment]
Since this might interest others, this is now possible with a little work from your side. Check section `How to change the video frames on videos` in https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md
